cve_bin_tool/parsers/rpm.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import io
+from enum import IntEnum
+
+from cve_bin_tool.parsers import Parser
+
+
+class RpmParser(Parser):
```

Comment:
https://github.com/srossross/rpmfile also contains a parser if you wanted to reuse anything from that.

https://github.com/srossross/rpmfile/blob/185ac915eb701d15bcb2338ad63ca81366f4ce16/rpmfile/rpmdefs.py#L11-L12

Reply:
i just took a look at the lib, and i dont know..

i thought there would be more exception handling, and validation of headers, but its just going over to the next header, and next, not even taking into account 8 byte alignment.. and just searching for the magic pattern, which shouldn't pass as the structure and offsets are well known from fields, i get the feeling that the malformed rpm could pass through it..

i could switch to it, since its used in the project, but i got my doubts about it being strict verifier of rpm